src/mlpack/core/util/io.cpp

```diff
@@ -267,3 +267,39 @@ void IO::ClearSettings()
   GetSingleton().aliases = persistentAliases;
   GetSingleton().functionMap = persistentFunctions;
 }
+
+void IO::CheckInputMatrices()
+{
+  typedef typename std::tuple<data::DatasetInfo, arma::mat> TupleType;
+  std::map<std::string, util::ParamData>::iterator itr;
+
+  for (itr = IO::Parameters().begin(); itr != IO::Parameters().end(); ++itr)
+  {
+    std::string paramName = itr->first;
+    std::string paramType = itr->second.cppType;
+
+    if (paramType == "arma::mat")
+      IO::CheckInputMatrix<arma::Mat<double>>(paramName);
+    else if (paramType == "arma::Mat<size_t>")
+      IO::CheckInputMatrix<arma::Mat<size_t>>(paramName);
+    else if (paramType == "arma::colvec")
+      IO::CheckInputMatrix<arma::Col<double>>(paramName);
+    else if (paramType == "arma::Col<size_t>")
+      IO::CheckInputMatrix<arma::Col<size_t>>(paramName);
+    else if (paramType == "arma::rowvec")
+      IO::CheckInputMatrix<arma::Row<double>>(paramName);
+    else if (paramType == "arma::Row<size_t>")
+      IO::CheckInputMatrix<arma::Row<size_t>>(paramName);
+    else if (paramType == "std::tuple<data::DatasetInfo, arma::mat>")
+    {
+      std::string errMsg1 = "The input " + paramName + " has NaN values.";
+      std::string errMsg2 = "The input " + paramName + " has inf values.";
+
+      if (std::get<1>(GetParam<TupleType>(paramName)).has_nan())
+        Log::Fatal << errMsg1 << std::endl;
+      if (std::get<1>(GetParam<TupleType>(paramName)).has_inf())
+        Log::Fatal << errMsg2 << std::endl;
```

Comment:
You could use `IO::CheckInputMatrix()` here too, just use `std::get<1>(GetParam<TupleType>(paramName))` and refactor `CheckInputMatrix()` to take both a `T` and `identifier`.  Basically you would call `GetParam` externally, and then `CheckInputMatrix()` would operate on the matrix it was given.  Let me know what you think---I think it's a nice simplification. :+1:

Reply:
Yes I feel that will be more generalized and will simplify the code. I have done the same.